coalib/results/Result.py

```diff
@@ -63,6 +65,13 @@ def __init__(self,
         if severity not in RESULT_SEVERITY.reverse:
             raise ValueError("severity is not a valid RESULT_SEVERITY")
 
+        self.diffs = {}
+        if(diffs and isinstance(diffs, dict)):
+            for diff in diffs:
+                self.diffs[diff.filename] = diff
```

Comment:
`self.diffs = {d.filename: d for d in diffs}` Simpler to read.
Also, this is if diffs is a dictionary ? Shouldnt it be when diffs is not a dictionary ?